app/controllers/results_controller.rb

```diff
@@ -272,8 +272,10 @@ def run_tests
                                                 submission.assignment.id,
                                                 [submission.grouping.group_id])
     session[:job_id] = @current_job.job_id
-    flash_message(:notice, I18n.t('automated_tests.tests_running'))
-    redirect_back(fallback_location: root_path)
+    flash_message(:success, I18n.t('automated_tests.tests_running'))
+    respond_to do |format|
```

Comment:
You can remove this block---it's not wrong, but our code base doesn't generally restrict the request types unless we need to distinguish between multiple format types.